### PR TITLE
Unique network flows use-case [10552]

### PR DESCRIPTION
**Pull request by MiguelCompany**
Adding documentation on the Unique Network flows use case.

Related to eProsima/Fast-DDS#1772

### CODE REVIEW

**Comment by richiprosima** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/746/)

File: docs/fastdds/use_cases/use_cases.rst [hunk 1]
```diff
@@ -63,6 +63,10 @@ with distributed systems:
   Under certain constraints, *Fast DDS* can provide application level communication
   between publishing and subscribing nodes avoiding any data copy during the process.
 
++ :ref:`use-case-unique-flows`.
+  APIs allowing the identification of the network flows being used, and requesting them to be unique,
+  are explained here.
```
Comment (EduPonz) [inline docs/fastdds/use_cases/use_cases.rst:68]: Although the sentence is completely correct, I'd go for something more direct (not accounting for line length):
```suggestion
  This use case illustrates the APIs that allow for the request of unique network flows, and for the identification of those in use.
```

File: docs/fastdds/use_cases/unique_netwotk_flows/unique_netwotk_flows.rst [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+.. contents::
+    :local:
+    :backlinks: none
+    :depth: 1
+
+Background
+----------
+
+IP networking is the pre-dominant inter-networking technology used nowadays.
+Ethernet, WiFi, 4G/5G telecommunication all rely on IP networking.
```
Comment (EduPonz) [inline docs/fastdds/use_cases/unique_netwotk_flows/unique_netwotk_flows.rst:21]:
```suggestion
Ethernet, WiFi, 4G/5G telecommunication, all of them rely on IP networking.
```

File: docs/fastdds/use_cases/unique_netwotk_flows/unique_netwotk_flows.rst [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+Streams of IP packets from a given source to destination are called *packet flows* or simply *flows*.
+The network QoS of a flow can be configured when using certain networking equipment (routers, switches).
+Such equipments are typically supporting 3GPP/5QI protocols to assign certain Network QoS parameters to
```
Comment (EduPonz) [inline docs/fastdds/use_cases/unique_netwotk_flows/unique_netwotk_flows.rst:25]: Is it tipically or is it more may support?
```suggestion
Such pieces of equipment typically support 3GPP/5QI protocols to assign certain Network QoS parameters to
```

File: docs/fastdds/use_cases/unique_netwotk_flows/unique_netwotk_flows.rst [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+specific flows.
+
```
Comment (EduPonz) [inline docs/fastdds/use_cases/unique_netwotk_flows/unique_netwotk_flows.rst:27]:
```suggestion
```

File: docs/fastdds/use_cases/unique_netwotk_flows/unique_netwotk_flows.rst [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+Requesting a specific Network QoS is usually done on the endpoint sending the data, as it is the one
+that usually haves complete information about the network flow.
+
+Applications may need to use specific Network QoS parameters on different topics.
+
+This means an application should be able to
```
Comment (EduPonz) [inline docs/fastdds/use_cases/unique_netwotk_flows/unique_netwotk_flows.rst:33]:
```suggestion
This means an application should be able to:
```

File: docs/fastdds/use_cases/unique_netwotk_flows/unique_netwotk_flows.rst [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+Definitions
+^^^^^^^^^^^
+
+**Network flow**: A tuple of networking resources selected by the middleware for transmission of messages from a
+DataWriter to a DataReader, namely:
+
+  - transport protocol: UDP or TCP
+  - transport port
+  - internet protocol: IPv4 or IPv6
+  - internet address
```
Comment (EduPonz) [inline docs/fastdds/use_cases/unique_netwotk_flows/unique_netwotk_flows.rst:55]:
```suggestion
  - Transport protocol: UDP or TCP
  - Source port
  - Internet protocol: IPv4 or IPv6
  - IP address
```

File: docs/fastdds/use_cases/unique_netwotk_flows/unique_netwotk_flows.rst [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+
+**Network Flow Endpoint (NFE)**: The portion of a network flow specific to the DataWriter or the DataReader.
+In other words, each network flow has two NFEs; one for the DataWriter and the other for the DataReader.
```
Comment (EduPonz) [inline docs/fastdds/use_cases/unique_netwotk_flows/unique_netwotk_flows.rst:58]:
```suggestion
In other words, each network flow has two NFEs; one for the DataWriter, and the other for the DataReader.
```

File: docs/fastdds/use_cases/unique_netwotk_flows/unique_netwotk_flows.rst [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+
+APIs
+^^^^
+
+Fast DDS provides the APIs needed to get the list of NFEs used by a DataWriter or a DataReader.
```
Comment (EduPonz) [inline docs/fastdds/use_cases/unique_netwotk_flows/unique_netwotk_flows.rst:63]:
```suggestion
Fast DDS provides the APIs needed to get the list of NFEs used by a given DataWriter or a DataReader.
```

File: docs/fastdds/use_cases/unique_netwotk_flows/unique_netwotk_flows.rst [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+
+On the DataWriter, |DataWriter::get_sending_locators-api| allows the application to obtain the list of locators
+from which the writer may send data.
+
+On the DataReader, |DataReader::get_listening_locators-api| allows the application to obtain the list of locators
+on wich the reader is listening.
```
Comment (EduPonz) [inline docs/fastdds/use_cases/unique_netwotk_flows/unique_netwotk_flows.rst:69]: Maybe we can put this into a list

File: docs/fastdds/use_cases/unique_netwotk_flows/unique_netwotk_flows.rst [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+
+Requesting unique flows
+-----------------------
+
+A unique flow can be created just ensuring that one of the two NFEs are unique.
```
Comment (EduPonz) [inline docs/fastdds/use_cases/unique_netwotk_flows/unique_netwotk_flows.rst:74]:
```suggestion
A unique flow can be created by ensuring that at least one of the two NFEs are unique.
```

File: docs/fastdds/use_cases/unique_netwotk_flows/unique_netwotk_flows.rst [hunk 11]
```diff
@@ -0,0 +1,82 @@
...
+On Fast DDS, there are two ways to select unique listening locators on the DataReader.
```
Comment (EduPonz) [inline docs/fastdds/use_cases/unique_netwotk_flows/unique_netwotk_flows.rst:75]:
```suggestion
On Fast DDS, there are two ways to select unique listening locators on the DataReader:
```

File: docs/fastdds/use_cases/unique_netwotk_flows/unique_netwotk_flows.rst [hunk 12]
```diff
@@ -0,0 +1,82 @@
...
+
+* The application can specify on which locators the DataReader should be listening, using :ref:`rtpsendpointqos` on
```
Comment (EduPonz) [inline docs/fastdds/use_cases/unique_netwotk_flows/unique_netwotk_flows.rst:77]:
```suggestion
* The application can specify on which locators the DataReader should be listening. This is done using :ref:`rtpsendpointqos` on
```

File: docs/fastdds/use_cases/unique_netwotk_flows/unique_netwotk_flows.rst [hunk 13]
```diff
@@ -0,0 +1,82 @@
...
+  the :ref:`dds_layer_subscriber_dataReaderQos`.
+  In this case is the responsibility of the application to ensure the uniqueness of the locators used.
+* The application can request the reader to be created with unique listening locators, using a
```
Comment (EduPonz) [inline docs/fastdds/use_cases/unique_netwotk_flows/unique_netwotk_flows.rst:80]:
```suggestion
* The application can request the reader to be created with unique listening locators. This is done using a
```

**Review by EduPonz — CHANGES_REQUESTED** [review]
In general, all the description is very readable and understandable. However, I'd appreciate some code snippets using the feature. Not only will it help readers, but the code will also be verified. This way, the use-case would be more in accordance with the rest of them,

**Comment by richiprosima** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/747/)

**Comment by richiprosima** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/748/)

**Comment by richiprosima** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/750/)

**Comment by richiprosima** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/752/)

**Comment by richiprosima** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/753/)

**Comment by richiprosima** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/754/)

**Comment by richiprosima** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/755/)

**Comment by richiprosima** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/756/)

**Review by EduPonz — DISMISSED** [review]
LGTM with green CI 🚀

**Comment by richiprosima** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/757/)

**Comment by richiprosima** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/758/)

**Comment by richiprosima** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/759/)

**Comment by richiprosima** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/760/)

**Comment by richiprosima** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/761/)

**Comment by richiprosima** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/762/)

**Review by EduPonz — APPROVED** [review]
LGTM

**Pull request merged**